compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -2296,11 +2323,13 @@ impl<'tcx> Ty<'tcx> {
         matches!(self.kind(), Infer(_))
     }
 
+    /// Checks if this type is [`core::marker::PhantomData`].
     #[inline]
     pub fn is_phantom_data(self) -> bool {
         if let Adt(def, _) = self.kind() { def.is_phantom_data() } else { false }
     }
 
+    /// Checks if this type is a boolean.
```

Comment:
I appreciate you taking the time to document all these methods, but some like this one are somewhat self-explanatory and don't know how much value there is in documenting them. When you were working and navigating this codebase, was the lack of documentation on the `is_*` and `expect_` methods a hindrance?

Reply:
No, I have just added them since I was already documenting that file anyway.

I also thought that, in the future, `#[warn(missing_docs)]` or even `#[deny(missing_docs)]`could be added, ensuring documentation is always present.  In such case, even small function would need documentation.